mypy/type_visitor.py

```diff
@@ -347,3 +349,121 @@ def query_types(self, types: Iterable[Type]) -> T:
                 self.seen_aliases.add(t)
             res.append(t.accept(self))
         return self.strategy(res)
+
+
+class TypeQueryBool(SyntheticTypeVisitor[bool]):
```

Comment:
Related: #9602

Reply:
thanks for the headsup, I added the decorator for `allow_interpreted_subclasses=True`